python/packages/ai/teams/app.py

```diff
@@ -664,114 +664,116 @@ async def on_turn(self, context: TurnContext):
-
-            await state.load(context, self._options.storage)
-            state.temp.input = context.activity.text
-
-            # authentication
-            if (
-                self.options.auth
-                and self._auth
-                and (
-                    (
-                        (
-                            isinstance(self.options.auth.auto, bool)
-                            and self.options.auth.auto is True
-                        )
-                        or (
-                            callable(self.options.auth.auto)
-                            and self.options.auth.auto(context) is True
-                        )
-                    )
-                    or IN_SIGN_IN_KEY in state.user
-                )
-            ):
-                key: Optional[str] = (
-                    state.user[IN_SIGN_IN_KEY]
-                    if IN_SIGN_IN_KEY in state.user
-                    else self.options.auth.default
-                )
+            await self._start_typing(context)
+
+            self._remove_mentions(context)
+
+            state = await self._initialize_state(context)
+
+            if not await self._authenticate_user(context, state):
+                return
+
+            if not await self._run_before_turn_middleware(context, state):
+                return
+
+            await self._handle_file_downloads(context, state)
+
+            is_ok, matches = await self._on_activity(context, state)
+            if not is_ok:
+                await state.save(context, self._options.storage)
+                return
+            if matches == 0:
+                if not await self._run_ai_chain(context, state):
+                    return
+
+            if not await self._run_after_turn_middleware(context, state):
+                return
+
+            await state.save(context, self._options.storage)
+        except ApplicationError as err:
+            await self._on_error(context, err)
+        finally:
+            self.typing.stop()
+
+    async def _start_typing(self, context: TurnContext):
+        if self._options.start_typing_timer:
+            await self.typing.start(context)
+
+    def _remove_mentions(self, context: TurnContext):
+        if self.options.remove_recipient_mention and context.activity.type == ActivityTypes.message:
+            context.activity.text = context.remove_recipient_mention(context.activity)
+
+    async def _initialize_state(self, context: TurnContext):
+        state = cast(StateT, await TurnState.load(context, self._options.storage))
+
+        if self._turn_state_factory:
+            state = await self._turn_state_factory(context)
+
+        await state.load(context, self._options.storage)
+        state.temp.input = context.activity.text
+        return state
+
+    async def _authenticate_user(self, context: TurnContext, state):
+        if self.options.auth and self._auth:
+            auth_condition = (isinstance(self.options.auth.auto, bool)) or (
```

Comment:
need to also check

`and self.options.auth.auto is True`
`and self.options.auth.auto(context) is True`